File: gluecodium/src/test/resources/smoke/external_types/output/android/com/example/smoke/ExternalInterfaceImpl.java
/*
 *

 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
class ExternalInterfaceImpl extends NativeBase implements ExternalInterface {
    protected ExternalInterfaceImpl(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    public native void someMethod(final byte someParameter);
    @NonNull
    public native String getSomeProperty();
}

File: gluecodium/src/test/resources/smoke/external_types/output/android-kotlin/com/example/smoke/ExternalInterfaceImpl.kt
/*

 *
 */

@file:JvmName("ExternalInterfaceImpl")

package com.example.smoke

import com.example.NativeBase

/**
 * @suppress
 */
class ExternalInterfaceImpl : NativeBase, ExternalInterface {
    protected constructor(nativeHandle: Long, tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}

    override external fun someMethod(someParameter: Byte) : Unit

    override val someProperty: String
        external get



    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}
